### PR TITLE
Document COVERAGE_RCFILE

### LINKED ISSUE #3574: Document `COVERAGE_RCFILE`
Document the supported environment variable `COVERAGE_RCFILE` (location moved in #3571), next to  https://docs.cocotb.org/en/stable/building.html#envvar-COVERAGE

### PR DESCRIPTION
**Pull request by cmarqu**
Closes #3574.

### CODE REVIEW

**Comment by ktbarrett** [comment]
Not sure we need to document `COCOTB_LIBRARY_COVERAGE` as it's for our regressions only.

File: docs/source/building.rst [hunk 1]
```diff
@@ -167,10 +167,36 @@ Regression Manager
...
+.. envvar:: COCOTB_LIBRARY_COVERAGE
+
+    Enable to collect Python coverage data for code in cocotb's own library.
+    Branch coverage is collected
+    and only files in the cocotb package directory are included.
+
+    This needs the :mod:`coverage` Python module to be installed.
+
+    .. versionadded:: 1.4
+
```
Comment (ktbarrett) [inline docs/source/building.rst:199]: Do we need to document this? Would users be interested in this?

Comment (cmarqu) [reply docs/source/building.rst]: I removed it now.

File: docs/source/building.rst [hunk 2]
```diff
@@ -167,10 +167,26 @@ Regression Manager
 
 .. envvar:: COVERAGE
 
-    Enable to report Python coverage data. For some simulators, this will also report :term:`HDL` coverage.
+    Enable to collect Python coverage data for user code.
+    For some simulators, this will also report :term:`HDL` coverage.
```
Comment (ktbarrett) [inline docs/source/building.rst:171]: Is this true?

**Review by ktbarrett — APPROVED** [review]

File: docs/source/building.rst (hunk 2 above)
Comment (cmarqu) [reply docs/source/building.rst:171]: It's true for Riviera: https://github.com/cocotb/cocotb/blob/17038c0ff372423f64282146a90c8d73cf021b8d/src/cocotb_tools/makefiles/simulators/Makefile.riviera#L87

Came in 11 years ago in https://github.com/cocotb/cocotb/commit/71550332dfa4e5bf6a3dd3ec64a2241cdb0dadd1

I had actually planned to deprecate this together with other envvars in #1495; 1.9/2.0 seems like a good point in time to finally do it.

**Pull request merged**